### PR TITLE
DEV-2922: Add TAS autocomplete

### PR DESCRIPTION
**Pull request by kbard**
**Description**

In order to support Treasury Account Symbol (TAS) and Federal Account (FA) filtering in the advanced search, we are going to provide a TAS/FA builder on the website that will allow users to enter TAS/FA components for searching.

As with other filters on the site, we will be implementing an autocomplete feature to assist users with building these filters.  Each component provided in the TAS/FA can be used to filter other TAS/FA components in the typeahead to narrow available choices.

This is the endpoint to support the autocomplete functionality.

**Technical details**

Added 7 new endpoints:
* `/api/v2/autocomplete/accounts/ata/` - Allocation Transfer Agency Identifier (ATA)
* `/api/v2/autocomplete/accounts/aid/` - Agency Identifier (AID)
* `/api/v2/autocomplete/accounts/bpoa/` - Beginning Period of Availability (BPOA)
* `/api/v2/autocomplete/accounts/epoa/` - Ending Period of Availability (EPOA)
* `/api/v2/autocomplete/accounts/a/` - Availability Type Code (A)
* `/api/v2/autocomplete/accounts/main/` - Main Account Code (MAIN)
* `/api/v2/autocomplete/accounts/sub/` - Sub-Account Code (SUB)

**Requirements for PR merge**

1. [x] Unit & integration tests updated
2. [x] API documentation/contracts are part of pull request https://github.com/fedspendingtransparency/usaspending-api/pull/1874
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Frontend <OPTIONAL>
4. [x] No materialized views need changing... yet
5. [x] Front end work is part of another, not-yet-submitted pull request
6. [x] Data validation completed
7. [x] No Operations ticket required
8. [x] Jira Ticket [DEV-2922](https://federal-spending-transparency.atlassian.net/browse/DEV-2922):
    - [x] Link to this Pull-Request

### CODE REVIEW

File: usaspending_api/references/tests/test_tas_autocomplete.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+BASE_ENDPOINT = "/api/v2/autocomplete/accounts/"
+
+
+class TASAutocompleteTestCase(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        mommy.make("references.CGAC", cgac_code="000", agency_name="Agency 000", agency_abbreviation="A000")
+        mommy.make("references.CGAC", cgac_code="002", agency_name="Agency 002", agency_abbreviation="A002")
```
Comment (tony-sappe) [inline usaspending_api/references/tests/test_tas_autocomplete.py]: Could we have a test case to make sure a string that doesn't start but ends with the search text isn't found?

File: usaspending_api/references/v2/views/tas_autocomplete.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            elif current_column == requested_column:
+                kwargs[current_column + "__startswith"] = current_value
+            else:
+                kwargs[current_column] = current_value
+
+        results = (
+            TreasuryAppropriationAccount.objects.filter(**kwargs)
+            .values_list(requested_column, flat=True)
+            .distinct()
+            .order_by(requested_column)[:limit]
```
Comment (tony-sappe) [inline usaspending_api/references/v2/views/tas_autocomplete.py]: Alphabetical ASC should work, was there any discussion around ordering in the mock?

File: usaspending_api/references/tests/test_tas_autocomplete.py (hunk 1 above)
Comment (kbard) [reply usaspending_api/references/tests/test_tas_autocomplete.py]: 👍

File: usaspending_api/references/v2/views/tas_autocomplete.py (hunk 2 above)
Comment (kbard) [reply usaspending_api/references/v2/views/tas_autocomplete.py]: Checked with the UX designer and he is good with alphabetical (pseudo-numeric) sorting.

**Pull request merged**